source/common/buffer/buffer_impl.h

```diff
@@ -749,6 +753,30 @@ class OwnedImpl : public LibEventInstance {
   void commit(uint64_t length, absl::Span<RawSlice> slices,
               ReservationSlicesOwnerPtr slices_owner) override;
 
+  uint8_t* inlineReserve(uint64_t size) override {
+    // If no any slice in the buffer then create one new slice based on the expected continuous
+    // memory size.
+    if (slices_.empty()) {
+      slices_.emplace_back(Slice(size, account_));
+      length_ += size;
+      slices_.back().reservable_ += size;
+      return slices_.back().base_;
+    }
+
+    // If the back slice has enough continuous memory then reserve the memory segment and mark it
+    // as already used.
+    auto& back = slices_.back();
```

Comment:
can we refrain from using `auto` where the type is not obvious from immediate context?

Reply:
get it.